app/helpers/pagelets_helper.rb

```diff
@@ -3,33 +3,39 @@ def pagelets_for(mountpoint)
     Pagelets::Manager.sorted_pagelets_at("#{controller_name}/#{action_name}", mountpoint)
   end
 
-  def render_pagelets_for(mountpoint, opts = {})
+  def render_pagelets_for(mountpoint, model, opts = {})
     result = ""
     pagelets_for(mountpoint).each do |pagelet|
-      result += render_pagelet(pagelet, opts)
+      result += render_pagelet(pagelet, model, opts)
     end
     result.html_safe
   end
 
-  def render_tab_content_for(mountpoint, opts = {})
+  def render_tab_content_for(mountpoint, model, opts = {})
     result = ""
     pagelets_for(mountpoint).each do |pagelet|
+      next unless pagelet.onlyif.call model
       result += "<div id='#{pagelet.id}' class='tab-pane'>"
-      result += render_pagelet(pagelet, opts)
+      result += render_pagelet(pagelet, model, opts)
       result +=  "</div>"
     end
     result.html_safe
   end
 
-  def render_tab_header_for(mountpoint, opts = {})
+  def render_tab_header_for(mountpoint, model, opts = {})
     result = ""
     pagelets_for(mountpoint).each do |pagelet|
+      next unless pagelet.onlyif.call model
```

Comment:
This could be an array so multiple subjects are supported, but that's only nice to have...